tests/test_common.c

```diff
@@ -295,3 +300,91 @@ test_helper_create_state_file (const char *name,
 
 	return true;
 }
+
+/**
+ * Run a VM that can be used to test qmp
+ *
+ * \param[out] socket_path qmp socket path
+ *
+ * \return qemu's pid on success, else -1
+ */
+pid_t run_qmp_vm(char **socket_path) {
+	struct stat st;
+	pid_t pid;
+	int err_pipe[2] = { -1, -1 };
+	char tmpdir[] = "/tmp/.qmpXXXXXX";
+	char err;
+	ssize_t bytes;
+	pid_t ret = -1;
+
+	if (! mkdtemp (tmpdir)) {
+		fprintf (stderr, "failed to create tmp directory");
+		goto out;
+	}
+
+	*socket_path = g_strdup_printf ("%s/hypervisor.sock", tmpdir);
+
+	if (pipe2(err_pipe, O_CLOEXEC) < 0) {
+		fprintf (stderr, "failed to create pipes");
+		return -1;
+	}
+
+	pid = fork ();
+	if (pid < 0) {
+		fprintf (stderr, "failed to fork parent");
+		return -1;
+	} else if (! pid) {
+		/* child */
+		int i = 0;
+		close (err_pipe[0]);
+
+		char **argv = calloc(12+1, sizeof(char *));
+		argv[i++] = g_strdup(QEMU_PATH);
+		argv[i++] = g_strdup("-m");
+		argv[i++] = g_strdup("1M");
+		argv[i++] = g_strdup("-qmp");
+		argv[i++] = g_strdup_printf("unix:%s,server,nowait", *socket_path);
+		argv[i++] = g_strdup("-nographic");
+		argv[i++] = g_strdup("-vga");
+		argv[i++] = g_strdup("none");
+		argv[i++] = g_strdup("-net");
+		argv[i++] = g_strdup("none");
+		argv[i++] = g_strdup("-smp");
+		argv[i++] = g_strdup("1");
+
+		g_free (*socket_path);
+
+		if (execvp (argv[0], argv) < 0) {
+			fprintf (stderr, "failed to exec child %s: %s",
+					argv[0],
+					strerror (errno));
+		}
+
+		write (err_pipe[1], "E", 1);
+		exit (EXIT_FAILURE);
+	}
+
+	/* parent */
+	close (err_pipe[1]);
+	err_pipe[1] = -1;
+	bytes = read (err_pipe[0], &err, sizeof(err));
+	if (bytes > 0) {
+		fprintf (stderr, "failed to exec hypervisor");
+		goto out;
+	}
+	/* waiting for qmp socket */
+	do {}while (stat (*socket_path, &st) != 0);
```

Comment:
This is a neat piece of code, but it'll spin forever if the hypervisor fails to create the socket so I think we need some sort of timeout here.